### ISSUE TITLE
Reroutes to _/ within AUT when setting location.replace(hash)

### ISSUE THREAD

**Issue by urbicon** [cypress-io/cypress#7488]
This is a follow up of #5629 with simpler and better reproducible example.

### Current behavior:

When 'replace' is executed on location with a hash only, this will cause cypress to open multiple iframes.

```javascript
location.replace('#test')
```

<img width="1904" alt="Bildschirmfoto 2020-05-26 um 22 31 54" src="https://user-images.githubusercontent.com/12167849/82947692-b8121d00-9fa0-11ea-930e-f8c95fa61300.png">

### Desired behavior:

Cypress should detect that the origin of the location did not change and stay in the current iframe.

### Test code to reproduce

```javascript
describe("Replacing Location Hash leads to Multiple Iframes", function() {
  before(function() {
    cy.visit('https://example.cypress.io')
  });
  it("Replaces the Location with a Hash", function() {
    cy.window().then(win => {
      win.location.replace('#test')
    });
  });
});
```

[Fork](https://github.com/urbicon/cypress-test-tiny/tree/5629-Route-not-working-after-hash-changed)  of cypress-test-tiny to reproduce the issue.

It might be a bit questionable whether this can be considered a bug, strictly speaking, as a single hash is not a fully qualified URL. But the desired behavior is working flawlessly in the major browsers and is also used in common frameworks like ExtJs.
Considering that the fix will probably be relatively simple, it would be good to have the current behavior fixed, as the problem is difficult to track down. It can also occur quite randomly, when navigation is happening quickly. 

### Versions

Cypress: 4.7.0
Browser: Chrome 83
OS: MacOS 10.15.4

**Comment by jennifer-shehane** [comment]
This issue still occurs with the `experimentalSourceRewriting` config set to true in 4.6.0+.

**Comment by flotwig** [comment]
Currently, `experimentalSourceRewriting` only rewrites code from the AUT, not spec code. So the `win.location.replace` from the original post would not be intercepted by the rewriting logic. 

Possibly this is something that should change in the future, so a `window` object from spec code is the same as a `window` object in the AUT.

@urbicon please clarify, are you calling `win.location.replace` from your spec code? Or is it being called from the app itself?

**Comment by urbicon** [comment]
@flotwig I am calling the replace from within the application, not the spec.

**Comment by SevanAMPS** [comment]
Hello, 

I Have the same issue in my application except its my application itself who generate the URL.

**Comment by flotwig** [comment]
In the meantime, using an absolute URL instead of a relative URL should work around this issue.

**Comment by flotwig** [comment]
Wait, @urbicon and @SevanAMPS, are you guys using the new `experimentalSourceRewriting` flag? It should fix this issue. I just realized you guys never said that you were using it. Try enabling `experimentalSourceRewriting`: https://docs.cypress.io/guides/references/experiments.html#Configuration

**Comment by SevanAMPS** [comment]
First of all, Thanks @flotwig for your interest on our issue.

Indeed, experimentalSourceRewriting seems to solve my issue. However, now I have a lot of errors in my consol like 'Uncaught ReferenceError: jQuery is not defined' therefore some elements is not visible since I change my Cypress.json parameters. 

Can you help me plz ?

**Comment by jennifer-shehane** [comment]
@SevanAMPS Please [open a new issue](https://github.com/cypress-io/cypress/issues/new) filling out the issue template with a reproducible example and/or screenshots and more info detailing the issue when you use `experimentalSourceRewriting` flag. It is experimental, so there may still be some underlying issues we'd love to address.

**Comment by jennifer-shehane** [comment]
@urbicon As Zach stated above, this situation will not rewrite the routes properly when changed from within the specfile. In your original issue where you're testing your site https://github.com/cypress-io/cypress/issues/5629 -we have a potential fix available for this starting in [`4.6.0`](https://docs-staging.cypress.io/guides/references/changelog.html#4-6-0) as an [experiment](https://on.cypress.io/experiments).Can you try setting this config option in your `cypress.json` or elsewhere:

```json
{
	"experimentalSourceRewriting": true
}
```

Please update the original issue with the result.
